### DISCUSSION TITLE
Integrating Storybook and 11ty

### DISCUSSION THREAD

**Discussion by pauleveritt** [11ty/eleventy#2164] [Uncategorized]
I'm trying to integrate Storybook and 11ty. Specifically, approaching my shortcode/Nunjucks development as "components", and component-driven development using stories and tests.

I got reasonably far but I think there's a fundamental issue: Storybook webpack wants to target the browser but 11ty wants to target node. I could drop back to pure Nunjucks with no 11ty but I was hoping to use 11ty for rendering, to then let me do shortcodes (and the entire 11ty config setup.)

Here's a README write up in a reproducible repo. Wonder if anybody has any thoughts?

https://github.com/pauleveritt/storybook_11ty

**Comment by pdehaan** [comment]
Saw this in Twitter today, not sure if it helps you, or future travelers https://multiline.co/mment/2022/01/eleventy-storybook/

**Comment by pauleveritt** [comment]
That's very helpful, thanks @pdehaan . I also tried that path, but macros aren't as attractive as shortcodes. Specifically, macro-components can't have any logic. Also, don't integrate as well into 11ty. I am currently trying the approach of having 11ty render Storybook JSON.

I might though decide the perfect is the enemy of the good and "leak" the component logic out of the macro.
